### PR TITLE
[ALLUXIO-2564] Improve entry/exit debug logging

### PR DESCRIPTION
**Pull request by calvinjia**
https://alluxio.atlassian.net/browse/ALLUXIO-2564

Follow up to #4798 which makes the logging more failure resilient.

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 1]
```diff
@@ -39,10 +39,35 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: can we implement `public static <T> T call(RpcCallable<T> callable)` 
by `public static <T> T call(Logger logger, RpcCallable<T> callable)`?

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 2]
```diff
@@ -39,10 +39,35 @@
...
+   * Calls the given {@link RpcCallable} and handles any exceptions thrown. This method also logs
+   * enter and exit when debug level logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   */
+  public static <T> T call(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: if this method is meant to be different from the other two overloading methods (in terms of what to log), perhaps use a different name. Overloading methods is supposed to be used when the logic is exactly the same.

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 3]
```diff
@@ -61,19 +86,51 @@
...
+   * method also logs enter and exit when debug level logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   * @throws ThriftIOException if the callable throws an IOException
+   */
+  public static <T> T call(Logger logger, RpcCallableThrowsIOException<T> callable)
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: NIT: I would probably put `logger` as the second parameter. This is minor, up to you

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 4]
```diff
@@ -39,10 +39,35 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
       throw e.toThrift();
     } catch (Exception e) {
-      LOG.error("Unexpected error running rpc", e);
+      LOG.error("Unexpected Error: {}", callable, e);
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: do we expect `IOException` thrown here? should we have a separate catch block for `IOException` if yes?

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 5]
```diff
@@ -39,10 +39,35 @@
...
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   */
+  public static <T> T call(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
+    try {
+      logger.debug("Enter: {}", callable);
+      T ret = callable.call();
+      logger.debug("Exit: {}", callable);
```
Comment (gpang) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Do we also want timing information about the call? Not having to do manual timestamp subtraction would help debugging.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I think this block violates our exception handling guidelines, which are:

either log the exception or re-throw it but not both

@aaudiber can you please comment

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 4 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: same comment as above

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 2 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I would change the above method to invoke this method with the static LOG

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: do you want to put the "Exit" log statement into a finally block?

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 6]
```diff
@@ -39,10 +39,35 @@
...
+      return ret;
+    } catch (AlluxioException e) {
+      logger.debug("Internal Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: again, I think we should either log the error or throw exception but not both

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 7]
```diff
@@ -39,10 +39,35 @@
...
+      throw e.toThrift();
+    } catch (Exception e) {
+      logger.error("Unexpected Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ditto

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 8]
```diff
@@ -61,19 +86,51 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ditto

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 9]
```diff
@@ -61,19 +86,51 @@
...
+      throw e.toThrift();
+    } catch (IOException e) {
+      LOG.debug("I/O Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ditto

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 10]
```diff
@@ -61,19 +86,51 @@
...
+      throw new ThriftIOException(e.getMessage());
+    } catch (Exception e) {
+      LOG.error("Unexpected Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ditto

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: the single argument `call` method should invoke this one with `LOG`

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 11]
```diff
@@ -61,19 +86,51 @@
...
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   * @throws ThriftIOException if the callable throws an IOException
+   */
+  public static <T> T call(Logger logger, RpcCallableThrowsIOException<T> callable)
+      throws AlluxioTException, ThriftIOException {
+    try {
+      logger.debug("Enter: {}", callable);
+      T ret = callable.call();
+      logger.debug("Exit: {}", callable);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: put into `finally` block?

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 12]
```diff
@@ -61,19 +86,51 @@
...
+      return ret;
+    } catch (AlluxioException e) {
+      logger.debug("Internal Error: {}", callable, e);
       throw e.toThrift();
     } catch (IOException e) {
-      LOG.warn("I/O error when running rpc", e);
+      logger.debug("I/O Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: log or throw exception but not both

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 13]
```diff
@@ -61,19 +86,51 @@
...
       throw new ThriftIOException(e.getMessage());
     } catch (Exception e) {
-      LOG.error("Unexpected error running rpc", e);
+      logger.error("Unexpected Error: {}", callable, e);
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ditto

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 14]
```diff
@@ -61,19 +86,51 @@
...
       throw new UnexpectedAlluxioException(e).toThrift();
     }
   }
 
   /**
-   * An interface representing a callable which can only throw Alluxio exceptions.
+   * An interface representing a callable which can only throw Alluxio exceptions. The toString
+   * method of classes implementing this interface should be in the format "CallName.
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: the `.` in the format string feels odd ... when I see `.`, I intuitively think "end of statement"

how about replacing `.` with `:` or alternatively `CallName(arg1:value1, arg2:value2, ...)`?

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I think I will take Bin's suggestion and make two separate methods.

The reason why I put `logger` as the first parameter is to make the method call sites look cleaner in terms of style. If you think having `logger` as the first argument is counterintuitive even if this method is not overloaded, I'm fine with changing it.

The only reason why `logger` needs to be passed in the first place is to have the log message print the actual class ie. `FileSystemMasterClientServiceHandler` as opposed to `RPCUtils`. I think its valuable but I just want to point out why I did so in case you guys think it is not worth the complexity in `RPCUtils`.

The reason why I cannot just pass `LOG` as the default argument is because we do not want enter/exit debug level logging for certain RPCs like `sessionHeartbeat` since those will be far too frequent.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: We need to log the exception here or the exception will only be thrown to the client and we will lose the server side stack trace.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I intended `Exit` to mean `Successful exit`, but I thought that was too verbose. We should see one `Enter` and one of `Exit/Error`. I thought seeing `Enter` then `Error` then `Exit` would be confusing.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I see, makes sense ... in that case, disregard all the instances of this comment

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: If you guys think it is valuable to have enter/exit logging for periodic RPCs like `sessionHeartbeat`, we could remove the non-logging version all together. However I just didn't think it would be practical (ie. 100 clients connected to a worker means ~200 log lines per second).

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: This case is unique in that it's the last place where we see the exception before it's sent over the wire. Client logs can be harder to track down, so at debug level I think it's reasonable to log and throw here.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 4 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: In this version of `call`, `IOException` is not expected to be thrown, so it is treated as an `Unexpected Error`

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: "Internal Error" suggests that something has gone wrong in the server, but this may not be the case, e.g. someone tries to create a file but it already exists. I'd prefer something more neutral like `Exception thrown by {}`.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I have two concerns:

1) enabling automated log parsing requires a simple and consistent logging -- right now, our logging couples "Enter" with "Exit" or "Internal Error" or "I/O Error" or "Unexpected Error", which seems unnecessarily complicated

2) readability -- I find not using "Exit" upon encountering an exception confusing as the methods indeed exits ... someone not familiar with the above convention might look for "Exit" in the logs

How about using something along the following lines:

- `("Enter: {}", callable)` upon method entry
- `("Exit (OK): {}", callable)` upon clean method exit  
- `("Exit (Error): {}", callable, e)` upon unexpected method exit

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 14 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: +1

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I agree that such excessive logging makes little sense ... if we wanted to get around that, we could supply a "nil" logger for RPCs we do not wish to log

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I like the `Exit (Error Type)` format, since we are explicitly categorizing the errors, I think it is valuable to propagate this to the user.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 4 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: @apc999 this method is for rpcs which don't throw `IOException`

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sgtm

Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: +1

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 9 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I think it would be fine to use the same "Exception running {}" for all of these exception logging statements. The name of the exception is already being logged along with the stack trace.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: If you guys are fine with a hardcoded check `if (LOG != logger)`, I think that will be the simplest solution. This should have minimal performance overhead, but there will be some added to the critical path.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 9 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I prefer having the category associated with the exception, so the user knows if there is something wrong within Alluxio, with I/O, or if something very bad has happened (unexpected). This information is hard to gather from just looking at the exception.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 14 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Makes sense, I prefer to use `:`

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Sure I can add timing info. There will be some performance overhead added to the critical path even without debug enabled, but if performance becomes an issue we can optimize afterward (ie. with `isDebugEnabled()` blocks).

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: works for me

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Would you be happier with the message if it was `Alluxio Error`? My thinking was to hint to the user which component is complaining (Alluxio or Non-Alluxio)

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 14 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: in that case, could we do:

`CallName: arg1=value1, arg2=value2, ...`

Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sg

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I think we're making two unsafe assumptions here:
- The user understands where we draw the distinction between Alluxio and Non-Alluxio errors.
- `IOException` always means Non-Alluxio error and non-`IOException` always means Alluxio error.

I like Jiri's proposal of using
```java
("Exit (Error): {}", callable, e)
```
to indicate that an error has occurred.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 9 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Resolved offline, this isn't our current contract so I will just use `Exit (Error)`.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I talked with @aaudiber some more and our categorization is not enforced/consistent so for now I will just do `Exit (OK)` and `Exit (Error)`. I think its nice to have a qualifier so you can search for errors easily.

Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sg

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sg

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: @gpang @aaudiber On second thought, could we do this in a separate PR? This change is already pretty large and I would like to isolate the timing info change when evaluating any performance impact.

**Comment by calvinjia** [comment]
@apc999 @aaudiber @gpang @jsimsa I've addressed most of the comments. For the timing info, I would prefer to do that in a separate PR since that will likely introduce performance overhead.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 5 above)
Comment (aaudiber) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sgtm

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 15]
```diff
@@ -17,63 +17,103 @@
...
+      throws AlluxioTException, ThriftIOException {
+    logger.debug("Enter: {}", callable);
+    T ret = call(logger, callable);
+    logger.debug("Exit (OK): {}", callable);
+    return ret;
+  }
+
+  /**
+   * An interface representing a callable which can only throw Alluxio exceptions. The toString
+   * method of classes implementing this interface should be in the format "CallName.
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: update format string

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 16]
```diff
@@ -88,7 +128,9 @@
   }
 
   /**
-   * An interface representing a callable which can only throw Alluxio or IO exceptions.
+   * An interface representing a callable which can only throw Alluxio or IO exceptions. The
+   * toString method of classes implementing this interface should be in the format "CallName.
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ditto

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 17]
```diff
@@ -17,63 +17,103 @@
...
+   */
+  public static <T> T callAndLog(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
+    logger.debug("Enter: {}", callable);
+    T ret = call(logger, callable);
+    logger.debug("Exit (OK): {}", callable);
+    return ret;
+  }
+
+  /**
+   * Calls the given {@link RpcCallableThrowsIOException} and handles any exceptions thrown. The
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: the javadoc for the format string seems to be duplicated here and in the RpcCallable interface ... I feel like we only need it in one of those places

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 18]
```diff
@@ -17,63 +17,103 @@
...
+   * implement a toString with the following format: "CallName: arg1=value1, arg2=value2,...".
+   * The toString will be used to log enter and exit information with debug logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   */
+  public static <T> T callAndLog(Logger logger, RpcCallable<T> callable) throws AlluxioTException {
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: so it seems like the difference between `call` and `callAndLog` is that `call` only logs errors, while callAndLog logs the entry and exit ... but it is confusing that the error exit points are logged in the `call` method

would the following make more sense:

```
  public static <T> T call(Logger logger, boolean logControlFlow, RpcCallable<T> callable) throws AlluxioTException {
    try {
      if (logControlFlow) {
        logger.debug("Enter: {}", callable);
      }
      T ret = callable.call();
      if (logControlFlow) {
        logger.debug("Exit (OK): {}", callable);
      }
    } catch (AlluxioException e) {
      if (logControlFlow) {
        logger.debug("Exit (Error): {}", callable);
      }
      logger.debug("Internal Alluxio error ...", e);
      throw e.toThrift();
    } catch (Exception e) {
      if (logControlFlow) {
        logger.debug("Exit (Error): {}", callable);
      }
      logger.error("Unexpected error ...", e);
      throw new UnexpectedAlluxioException(e).toThrift();
    }
  }
```

the point of doing the above being to separate the control flow logging (which is conditional) from error logging

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 19]
```diff
@@ -17,63 +17,103 @@
...
-      LOG.error("Unexpected error running rpc", e);
+      logger.error("Exit (Error): {}", callable, e);
       throw new UnexpectedAlluxioException(e).toThrift();
     }
   }
 
   /**
-   * An interface representing a callable which can only throw Alluxio exceptions.
+   * Calls the given {@link RpcCallable} and handles any exceptions thrown. The callable should
+   * implement a toString with the following format: "CallName: arg1=value1, arg2=value2,...".
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: the javadoc for the format string seems to be duplicated here and in the RpcCallable interface ... I feel like we only need it in one of those places

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 17 above)
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: Good catch, I think it is only required in the `callAndLog` method now, since `RpcCallables` can use `call` without overriding `toString`

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 18 above)
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: my main concern of the current implementation is that it allows for "Exit" log statements that do not have matching "Enter" statements

Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: @aaudiber and I also talked about this. We thought `Exit (Error)` made sense even without a corresponding `Enter` statement. If debug logging is enabled, we will always get matching `Enter/Exit`, and if it is disabled, the user will only see `Exit` statements which is reasonable since we are exiting, just with an error.

The upside of your proposed approach is that there is only 1 method and we get `Enter/Exit` matching logs. The downside of that approach is the method is fairly complicated and will get more complicated as we want to enrich the additional logging.

Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: ok, I still find the API (`callAndLog` vs `call` which also logs) unintuitive, but my main concerns for this change have been addressed so if you and @aaudiber are happy with it, then I won't stand in the way of progress :-)

File: core/server/master/src/main/java/alluxio/master/block/BlockMasterClientServiceHandler.java [hunk 20]
```diff
@@ -52,7 +56,7 @@ public long getServiceVersion() {
 
   @Override
   public List<WorkerInfo> getWorkerInfoList() throws AlluxioTException {
-    return RpcUtils.call(new RpcCallable<List<WorkerInfo>>() {
+    return RpcUtils.call(LOG, new RpcCallable<List<WorkerInfo>>() {
```
Comment (jsimsa) [inline core/server/master/src/main/java/alluxio/master/block/BlockMasterClientServiceHandler.java]: how do you decide when to use `call` vs `callAndLog`?

Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/block/BlockMasterClientServiceHandler.java]: Everything which is not automatic/periodic should eventually use `callAndLog`, currently I am only targeting the 3 most used service handlers and based on the usefulness of this approach, we can  expand the use to other handlers.

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 3 above)
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sg

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 4 above)
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: sg

File: core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java [hunk 21]
```diff
@@ -279,100 +305,122 @@ public String call() throws AlluxioException {
...
+            options);
+      }
     });
-    LOG.debug("Exit Mount. alluxioPath:{}, ufsPath:{}, options:{}", alluxioPath, ufsPath, options);
   }
 
   @Override
   public void remove(final String path, final boolean recursive)
       throws AlluxioTException, ThriftIOException {
     LOG.debug("Enter Remove. path:{}, recursive:{}", path, recursive);
```
Comment (ifcharming) [inline core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java]: sorry for the post-merge comment: but shall we remove this `LOG.debug`?

Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java]: @ifcharming Good catch, I did another pass to clean this up in #4808

File: keyvalue/server/src/main/java/alluxio/master/keyvalue/KeyValueMasterClientServiceHandler.java [hunk 22]
```diff
@@ -51,7 +57,7 @@ public long getServiceVersion() {
   @Override
   public void completePartition(final String path, final PartitionInfo info)
       throws AlluxioTException {
```
Comment (ParanoiaW) [inline keyvalue/server/src/main/java/alluxio/master/keyvalue/KeyValueMasterClientServiceHandler.java]: Shall we override toString() for these methods?

**Comment by calvinjia** [comment]
@ParanoiaW Yes, if using `callAndLog`, the callable should override `toString`
